front/components/assistant_builder/vaults/VaultSelector.tsx

```diff
@@ -41,64 +47,63 @@ export function VaultSelector({
       })
     )
     .flat();
-  // TODO: we are using Checkboxes here as our RadioButton isn't flexible
-  // enough to allow a onClick callback on a disabled item and to render
-  // elements in between labels. We are aiming to refactor RadioButton
+
   return (
     <>
-      {sortedVaults.map((vault, index) => {
-        const isDisabled =
-          allowedVaults && !allowedVaults.some((v) => v.sId === vault.sId);
-        const isChecked = selectedVault === vault.sId;
+      <RadioGroup
+        value={selectedVault}
+        onValueChange={(value) => setSelectedVault(value)}
+      >
+        {sortedVaults.map((vault, index) => {
+          const isDisabled =
+            allowedVaults && !allowedVaults.some((v) => v.sId === vault.sId);
 
-        return (
-          <div key={vault.sId}>
-            {index > 0 && <Separator />}
-            <div
-              className="flex items-center py-2"
-              onClick={() => {
-                if (isDisabled) {
-                  setAlertIsDialogOpen(true);
-                }
-              }}
-            >
-              <Checkbox
-                checked={isChecked}
-                onCheckedChange={() => {
-                  if (!isDisabled) {
-                    setSelectedVault(isChecked ? "" : vault.sId);
-                  }
-                }}
-                disabled={isDisabled}
-              />
-              <div className="flex items-center gap-2">
-                <Icon
-                  visual={getVaultIcon(vault)}
-                  size="md"
-                  className={classNames(
-                    "ml-3 mr-2 inline-block flex-shrink-0 align-middle",
-                    isDisabled ? "text-element-700" : "text-brand"
-                  )}
-                />
-                <span
-                  className={classNames(
-                    "font-bold",
-                    "align-middle",
-                    isDisabled ? "text-element-700" : "text-element-900"
-                  )}
+          return (
+            <>
```

Comment:
you shoud have a key here
```suggestion
            <React.Fragment key={vault.sId}>
```